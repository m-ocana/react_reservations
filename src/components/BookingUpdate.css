
.booking_update {
  flex: 1;
}

.title {
  font-size: 1rem;
}

form {
  border: 1px solid black;
}

form div {
  margin-bottom: 16px;
}

.label_for {
  font-weight: bold;
}

div span {
  display: block;
  margin-bottom: 1px;
  padding: 4px;
}

.booking_update p {
  margin: 8px 0;
  padding: 0 4px;
}

@media (max-width: 480px) {
  .booking_update {
    flex: 100%;
  }
}
