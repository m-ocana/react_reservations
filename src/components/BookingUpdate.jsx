import React, { Component, PropTypes } from 'react'
import Button from './Button'
import RadioButton from './RadioButton'
import { formatName } from '../helpers/utils'
import style from './BookingUpdate.css'

export default class BookingUpdate extends Component {

  static PropTypes = {
    booking: PropTypes.object,
    handleModalClose: PropTypes.func,
    handleBookingChange: PropTypes.func
  }

  static defaultProps = {
    booking: null,
    handleModalClose: () => {},
    handleBookingChange: () => {}
  }

  onRadioButtonChange = (e) => {
    this.props.handleBookingChange(e)
  }

  render() {
    const { booking, handleModalClose } = this.props
    if (!booking) return null
    const { cancelled, notes, seated, time, partySize } = booking
    return (
      <div className={style.booking_update}>
        <h2 className={style.title}>Booking update</h2>
        <form>
          <Button
            className="close"
            clickHandler={handleModalClose}
          />
          <div>
            <span className={style.label_for}>Name</span>
            <span>{formatName(booking)}</span>
          </div>
          <div>
            <span className={style.label_for}>Time</span>
            <span>{time}</span>
          </div>
          <div>
            <span className={style.label_for}>Covers</span>
            <span>{partySize}</span>
          </div>
          <div>
            <span className={style.label_for}>Seated</span>
            <RadioButton
              checked={!cancelled && !seated}
              label="No"
              name="selection"
              clickHandler={this.onRadioButtonChange}
              value="not_seated"
            />
            <RadioButton
              checked={!cancelled && seated}
              label="Yes"
              name="selection"
              clickHandler={this.onRadioButtonChange}
              value="seated"
            />
            <RadioButton
              checked={cancelled}
              label="Cancelled"
              name="selection"
              clickHandler={this.onRadioButtonChange}
              value="cancelled"
            />
          </div>
          { !!notes &&
            <div>
              <span className={style.label_for}>Notes</span>
              <p>{notes}</p>
            </div>
          }
        </form>
      </div>
    )
  }
}
