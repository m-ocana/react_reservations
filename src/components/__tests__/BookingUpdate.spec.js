import renderComponent from '../../helpers/test-component'
import BookingUpdate from '../BookingUpdate'

const renderBookingUpdate = renderComponent(BookingUpdate)
const initialProps = {
  booking: {
    title: 'Mr',
    firstName: 'Fred',
    lastName: 'Bloggs',
    time: '16.00',
    partySize: 2,
    seated: false,
    cancelled: false,
    notes: ''
  },
  handleModalClose: jest.fn(),
  handleBookingChange: jest.fn()
}
describe('<BookingUpdate /> component', () => {
  describe('@renders', () => {
    it('should not render if no booking is passed as prop', () => {
      expect(renderBookingUpdate().getTree())
        .toMatchSnapshot()
    })

    it('should render with booking passed as prop', () => {
      expect(renderBookingUpdate(initialProps).getTree())
        .toMatchSnapshot()
    })

    it('should render booking with note', () => {
      expect(renderBookingUpdate({
        ...initialProps,
        booking: {
          ...initialProps.booking,
          notes: 'By appointment of the royal family'
        }
      }).getTree()).toMatchSnapshot()
    })
  })

  describe('@events', () => {
    describe('on X button click', () => {
      it('should invoke handleModalClose function', () => {
        const { wrapper, instance } = renderBookingUpdate(initialProps)
        wrapper.find('.close').props().clickHandler()
        expect(instance.props.handleModalClose).toHaveBeenCalled()
      })
    })

    describe('on radio button change', () => {
      it('should trigger handleBookingChange', () => {
        const { wrapper, instance } = renderBookingUpdate({
          ...initialProps,
          seated: true
        })
        wrapper.find('RadioButton').first().props().clickHandler()
        expect(instance.props.handleModalClose).toHaveBeenCalled()
      })
    })
  })
})
