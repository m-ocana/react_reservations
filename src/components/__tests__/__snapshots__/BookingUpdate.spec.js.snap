// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookingUpdate /> component @renders should not render if no booking is passed as prop 1`] = `null`;

exports[`<BookingUpdate /> component @renders should render booking with note 1`] = `
<div
  className="booking_update"
>
  <h2
    className="title"
  >
    Booking update
  </h2>
  <form>
    <Button
      className="close"
      clickHandler={[Function]}
    />
    <div>
      <span
        className="label_for"
      >
        Name
      </span>
      <span>
        Mr Fred Bloggs
      </span>
    </div>
    <div>
      <span
        className="label_for"
      >
        Time
      </span>
      <span>
        16.00
      </span>
    </div>
    <div>
      <span
        className="label_for"
      >
        Covers
      </span>
      <span>
        2
      </span>
    </div>
    <div>
      <span
        className="label_for"
      >
        Seated
      </span>
      <RadioButton
        checked={true}
        className=""
        clickHandler={[Function]}
        label="No"
        name="selection"
        value="not_seated"
      />
      <RadioButton
        checked={false}
        className=""
        clickHandler={[Function]}
        label="Yes"
        name="selection"
        value="seated"
      />
      <RadioButton
        checked={false}
        className=""
        clickHandler={[Function]}
        label="Cancelled"
        name="selection"
        value="cancelled"
      />
    </div>
    <div>
      <span
        className="label_for"
      >
        Notes
      </span>
      <p>
        By appointment of the royal family
      </p>
    </div>
  </form>
</div>
`;

exports[`<BookingUpdate /> component @renders should render with booking passed as prop 1`] = `
<div
  className="booking_update"
>
  <h2
    className="title"
  >
    Booking update
  </h2>
  <form>
    <Button
      className="close"
      clickHandler={[Function]}
    />
    <div>
      <span
        className="label_for"
      >
        Name
      </span>
      <span>
        Mr Fred Bloggs
      </span>
    </div>
    <div>
      <span
        className="label_for"
      >
        Time
      </span>
      <span>
        16.00
      </span>
    </div>
    <div>
      <span
        className="label_for"
      >
        Covers
      </span>
      <span>
        2
      </span>
    </div>
    <div>
      <span
        className="label_for"
      >
        Seated
      </span>
      <RadioButton
        checked={true}
        className=""
        clickHandler={[Function]}
        label="No"
        name="selection"
        value="not_seated"
      />
      <RadioButton
        checked={false}
        className=""
        clickHandler={[Function]}
        label="Yes"
        name="selection"
        value="seated"
      />
      <RadioButton
        checked={false}
        className=""
        clickHandler={[Function]}
        label="Cancelled"
        name="selection"
        value="cancelled"
      />
    </div>
  </form>
</div>
`;
